mono/metadata/threads.c

```diff
@@ -4849,9 +4849,9 @@ self_abort_internal (void)
 
 	exc = mono_thread_request_interruption (TRUE);
 	if (exc)
-		mono_raise_exception (exc);
-
-	mono_thread_info_self_interrupt ();
+		mono_set_pending_exception (exc);
```

Comment:
I'll keep throwing then, it's really bad, but at least doesn't regress the behavior.